File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {

        int n = nums.size();
        vector<int> pre(n+1,1), post(n+1,1);
        

        pre[0] = nums[0];
        for(int i = 1 ; i < nums.size(); i++){
            pre[i] = pre[i-1] * nums[i];
        }   

        post[n-1] = nums[n-1];
        for(int i = n - 2; i >= 0 ; i--){
            post[i] =  post[i+1] * nums[i];
        }

        nums[0] = post[1];
        nums[n-1] = pre[n-2];
        for(int i = 1 ; i < n -1 ; i++){
            nums[i] = pre[i-1] * post[i+1];
        }   

        return nums;
    }
};

File: Data Structures & Algorithms/sort-an-array/submission-0.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        for i in range(1, len(nums)):
            key = nums[i]          # current element to insert
            j = i - 1

            # Move elements of nums[0..i-1] that are greater than key
            # to one position ahead of their current position
            while j >= 0 and nums[j] > key:
                nums[j + 1] = nums[j]
                j -= 1

            # Place key at after the element just smaller than it.
            nums[j + 1] = key

        return nums